server/vulnerabilities/db.go

```diff
@@ -21,37 +21,56 @@ func sqliteDB(dbPath string) (*sqlx.DB, error) {
 }
 
 func applyCPEDatabaseSchema(db *sqlx.DB) error {
+	// Use a new table cpe_2 containing new columns vendor, product. view cpe used for backwards compatibility
+	// with old fleet versions that use "select * from cpe ...". When creating the view, we need to
+	// select rowid because it is used for joins between the cpe and cpe_search tables
 	_, err := db.Exec(`
-	CREATE TABLE IF NOT EXISTS cpe (
-		cpe23 TEXT NOT NULL,
-		title TEXT NOT NULL,
-		version TEXT,
-		target_sw TEXT,
-		deprecated BOOLEAN DEFAULT FALSE
-	);
-	CREATE TABLE IF NOT EXISTS deprecated_by (
-		cpe_id INTEGER,
-		cpe23 TEXT NOT NULL,
-		FOREIGN KEY(cpe_id) REFERENCES cpe(rowid)
-	);
-	CREATE VIRTUAL TABLE IF NOT EXISTS cpe_search USING fts5(title, target_sw);
-	CREATE INDEX IF NOT EXISTS idx_version ON cpe (version);
-	CREATE INDEX IF NOT EXISTS idx_cpe23 ON cpe (cpe23);
-	CREATE INDEX IF NOT EXISTS idx_target_sw ON cpe (target_sw);
-	CREATE INDEX IF NOT EXISTS idx_deprecated_by ON deprecated_by (cpe23);
-	`)
+CREATE TABLE IF NOT EXISTS cpe_2 (
+    cpe23 TEXT NOT NULL,
+    title TEXT NOT NULL,
+    vendor TEXT,
+    product TEXT,
+    version TEXT,
+    target_sw TEXT,
+    deprecated BOOLEAN DEFAULT FALSE
+);
+CREATE VIEW IF NOT EXISTS cpe AS
+SELECT
+    rowid,
+    cpe23,
+    title,
+    version,
+    target_sw,
+    deprecated
+FROM cpe_2;
+CREATE TABLE IF NOT EXISTS deprecated_by (
+    cpe_id INTEGER,
+    cpe23 TEXT NOT NULL,
+    FOREIGN KEY(cpe_id) REFERENCES cpe(rowid)
+);
+CREATE VIRTUAL TABLE IF NOT EXISTS cpe_search USING fts5(title, target_sw);
+CREATE INDEX IF NOT EXISTS idx_cpe_2_cpe23 ON cpe_2 (cpe23);
+CREATE INDEX IF NOT EXISTS idx_cpe_2_vendor ON cpe_2 (vendor);
+CREATE INDEX IF NOT EXISTS idx_cpe_2_product ON cpe_2 (product);
+CREATE INDEX IF NOT EXISTS idx_cpe_2_version ON cpe_2 (version);
+CREATE INDEX IF NOT EXISTS idx_cpe_2_target_sw ON cpe_2 (target_sw);
+CREATE INDEX IF NOT EXISTS idx_deprecated_by ON deprecated_by (cpe23);
+`)
 	return err
 }
 
 func generateCPEItem(item cpedict.CPEItem) ([]interface{}, map[string]string, error) {
```

Comment:
Nit: named returns make sense here: `(cpes []interface{}, deprecations map[string]string, err error)`

Reply:
This kind of nit comments do not need to be addressed as this PR did not introduce the change. Am just sharing thoughts for future code like this.